### PR TITLE
Migrate `Backbone` validation to `zod`

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This migrates the validation of `Backbone` to `zod`.

## Changes

- Removes `BackboneValidator` and associated schema
- Adds `BackboneSchema` and infers type from it
- Add validation test

### CODE REVIEW

File: src/routes/chains/entities/backbone.entity.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class Backbone implements DomainBackbone {
   @ApiProperty()
   secure!: boolean;
   @ApiProperty()
-  settings!: Record<string, string>;
+  settings!: Record<string, unknown>;
```
Comment (iamacook) [inline src/routes/chains/entities/backbone.entity.ts:16]: The `settings` are [not only `string`s](https://safe-client.safe.global/v1/chains/1/about/backbone).

File: src/domain/backbone/entities/schemas/backbone.schema.ts [hunk 2]
```diff
@@ -1,24 +1,11 @@
...
-  required: ['name', 'version', 'api_version', 'secure', 'host'],
-};
+export const BackboneSchema = z.object({
+  name: z.string(),
+  version: z.string(),
+  api_version: z.string(),
+  secure: z.boolean(),
+  host: z.string(),
+  headers: z.array(z.string()),
+  settings: z.record(z.unknown()),
```
Comment (hectorgomezv) [inline src/domain/backbone/entities/schemas/backbone.schema.ts:10]: I think `headers` and `settings` are optional, as they are not included in the the JSON Schema `required` array.

**Review by hectorgomezv — CHANGES_REQUESTED** [review]
Great job! I've left a doubt/comment here: https://github.com/safe-global/safe-client-gateway/pull/1252#discussion_r1515733092

File: src/domain/backbone/entities/schemas/backbone.schema.ts (hunk 2 above)
Comment (iamacook) [reply src/domain/backbone/entities/schemas/backbone.schema.ts]: Good catch! I missed the `required` array. I've made them both `optional()` in 7b950b6552c8ad5faee24f60e042fe20c2bad7ef.

**Review by hectorgomezv — APPROVED** [review]

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**